src/data/samplers/modeMedian.js

```diff
@@ -0,0 +1,107 @@
+import d3 from 'd3';
+import {identity, noop} from '../../util/fn';
+
+export default function() {
+
+    var undefinedValue = d3.functor(undefined),
+        number = d3.functor(10),
+        accumulator = noop,
+        field = identity;
+
+    var modeMedian = function(data) {
+
+        if (number() > data.length) {
+            return data;
+        }
+
+        var minMax = getGlobalMinMax(data);
```

Comment:
d3 extent actually provides this for you: https://github.com/mbostock/d3/wiki/Arrays#d3_extent